### ISSUE TITLE
Project parts division. Part 2.1 Data preprocessing application on test set

### ISSUE THREAD

**Issue by AJ311021** [DSE511-Team-Avatar/Team-Avatar#12]
Need to preprocess the test set separately from the training set by applying all the same steps of cleaning, handling text and categorical data, and normalizing.

**Comment by isifletcher** [comment]
What do you mean by separately? We need to use what is obtained from the training set in order to normalize the testing set.

**Comment by AJ311021** [comment]
> What do you mean by separately? We need to use what is obtained from the training set in order to normalize the testing set.

I mean we need to do the split of the data first, and then do all the same preprocessing steps but separately on the test set. Especially normalizing all the data before splitting will cause data leakage (as it will be considering both train and test set at the same time) https://datascience.stackexchange.com/questions/54908/data-normalization-before-or-after-train-test-split/54909
In real life we will have test set that comes outside and we just apply all the preprocessing separately from the train set.

**Comment by isifletcher** [comment]
I added the code for this in my branch on the Jupyter notebook folder. Once we look over the final details I will create a .py file and we can close this issue.

**Comment by AJ311021** [comment]
I have uploaded files with the final version of the preprocessed data here (https://github.com/DSE511-Team-Avatar/Team-Avatar/tree/main/Jupyter_Notebooks) 

"Preprocessed data_FINAL.ipynb"

Please use this code for your modeling.
